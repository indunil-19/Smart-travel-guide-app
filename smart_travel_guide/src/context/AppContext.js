import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}