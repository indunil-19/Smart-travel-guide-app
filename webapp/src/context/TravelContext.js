import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 