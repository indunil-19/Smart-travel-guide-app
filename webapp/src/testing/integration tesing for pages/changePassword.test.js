import React from "react"
import {render,screen, cleanup,waitFor,fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import { createMemoryHistory } from 'history'
import ChangePassword from "../../screens/TravelPlanApp/changePassword"

describe('change password page', () => {

    const server=setupServer(
        rest.post("/user/updatepassword",(req,res,ctx)=>{
            return res(
                ctx.json({
                    prevoiusPassowrd:"1234",
                    newPassword:"4567"
                    })
            )
        }),
    )
 
    beforeEach(()=>{      
    })
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    afterEach(cleanup)


    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><ChangePassword/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><ChangePassword  /></TravelProvider></MemoryRouter>,div)
    })

    it("should return to my account page if password change sucessfully",()=>{
        const history = createMemoryHistory();
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><Router history={history}><TravelProvider><ChangePassword  /></TravelProvider></Router></MemoryRouter>,div)
        waitFor(()=>{
            fireEvent.change(screen.getByTestId("previous"), {target:{value:"1234"}})
            fireEvent.change(screen.getByTestId("new"), {target:{value:"4567"}})
            fireEvent.click(screen.getByTestId("submit"))
            waitFor(()=>{
                expect(history.location.pathname).toBe("/travelPlan/myAcccount")

            })
            
        })
    })

    

})
