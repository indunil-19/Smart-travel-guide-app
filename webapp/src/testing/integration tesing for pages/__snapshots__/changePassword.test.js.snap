// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change password page should match snapshot 1`] = `
<div
  className="css-4lqwvq"
>
  <h2
    className="chakra-heading css-0"
  >
    Change Password
  </h2>
  <div
    className="chakra-input__group css-16gncx0"
  >
    <div
      className="chakra-input__left-addon css-1lhl5nw"
    >
      previous password
    </div>
    <input
      className="chakra-input css-fu3wfu"
      data-testid="previous"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="previous password"
      type="text"
      value=""
    />
  </div>
  <div
    className="chakra-input__group css-16gncx0"
  >
    <div
      className="chakra-input__left-addon css-1lhl5nw"
    >
      New password
    </div>
    <input
      className="chakra-input css-fu3wfu"
      data-testid="new"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="New password"
      type="text"
      value=""
    />
  </div>
  <button
    className="chakra-button css-1pfig87"
    data-testid="submit"
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;
